chronos/apps/http/httpcommon.nim

```diff
@@ -0,0 +1,149 @@
+#
+#            Chronos HTTP/S common types
+#             (c) Copyright 2021-Present
+#         Status Research & Development GmbH
+#
+#              Licensed under either of
+#  Apache License, version 2.0, (LICENSE-APACHEv2)
+#              MIT license (LICENSE-MIT)
+import stew/results, httputils, strutils, uri
+import ../../asyncloop, ../../asyncsync
+import ../../streams/[asyncstream, boundstream]
+export results, httputils, strutils
+
+const
+  HeadersMark* = @[byte(0x0D), byte(0x0A), byte(0x0D), byte(0x0A)]
+  PostMethods* = {MethodPost, MethodPatch, MethodPut, MethodDelete}
+
+type
+  HttpResult*[T] = Result[T, string]
```

Comment:
an idea might be to move all result-based code to its own module so that it can be protected with `{.push: [Defect].}` to ensure no exceptions accidentally leak

Reply:
It would be tough to move all the procedures using `Result` to single module, but i will annotate all non-async procedures with `{.raises:[Defect].}`.

Reply:
So i marked most of the places, but there present some places which uses `HttpResult[T]` but could generate exceptions, because of `Future` usage.